agile-development/collaboration/readme.md

```diff
@@ -0,0 +1,71 @@
+
+# Collaboration
+
+## Why collaboration is important
+
+In engagements we aim to be highly collaborative because when we code together, we perform better, have a higher sprint velocity, and have a greater degree of knowledge sharing across the team.
+
+There are two common patterns we use for collaboration: Pairing and swarming.
+
+**Pair programming** (“pairing”) - two software engineers assigned to, and working on, one shared story at a time during the sprint. The Tech Lead assigns a user story to two engineers -- one primary engineer (story owner) and one secondary engineer (pairing assignee). The definition aligns with the Agile Alliance’s [definition](https://www.agilealliance.org/glossary/pairing).
+
+**Swarm programming** (“swarming”) - three or more software engineers collaborating on a high-priority item to bring it to completion.
+
+## How to pair program
+
+As mentioned, every story is intentionally assigned to a pair. The pairing assignee may be in the process of upskilling, nevertheless, they are equal partners in the development effort.
+Below are some general guidelines for pairing:
+
+- Upon assignment of the story/product backlog item (PBI), the pair needs to be deliberate about defining how to work together and have a firm definition of the work to be completed. This information should be expressed clearly in the story’s description and acceptance criteria. The expectations about this need to be communicated and agreed upon by both engineers and should be done prior to any actual working sessions.
+- The story owner and pairing assignee do not merely split the work up and sync regularly – they actively work together on the same tasks, often while sharing their screens via a Teams online session.
```

Comment:
Might be good to mention collaborative tools like [VS Live Share](https://visualstudio.microsoft.com/services/live-share/) here, which can be preferable to sharing screens. Not all collaboration needs to be screen-share based.

Reply:
Good point!